api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -164,6 +164,16 @@ public List<WorkspaceResponse> getPublishedWorkspaces() {
         .toList();
   }
 
+  @Override
+  public List<WorkspaceResponse> getFeaturedWorkspaces() {
+    return workspaceMapper
```

Comment:
Would it be better to  get all featuredWorkspace using Dao, grabbing the dao and using workspaceMapper on them?
If i understand this, this call is getting All workspaces from DB and then filtering the one that has featuredCategory == null

Reply:
The logic here is a bit subtle.  We need to combine 3 sources: AoU Workspace DB, AoU FeaturedWorkspace DB, and the Terra API.  We loop over the Terra workspaces here because `fireCloudService.getWorkspaces()` retrieves **all workspaces that the user can see in Terra** in a single call.  This will include all of the published workspaces because the user is a RT Reader, and we can distinguish which of these is Featured by making AoU DB calls.

Alternatively we could loop over the AoU FeaturedWorkspace DB entries (which would also give us AoU Workspace DB entry references) but then we'd have to make a choice about what to do in Terra: make the same List Workspaces call, or many individual Get Workspace calls?  I'd expect List Workspaces to be much faster.

I suppose we could do a hybrid of these approaches: get all Terra workspaces and all Featured Workspaces, and return the set of matches?

Reply:
Here is an attempt at that hybrid approach https://github.com/all-of-us/workbench/pull/8655